### PR TITLE
feat: add sortedSetFetchByScore

### PR DESCRIPTION
**Pull request by nand4011**
Add the sorted set fetch by score API, response types, integration tests, and documentation.

### CODE REVIEW

File: src/Cache/CacheClient.php [hunk 1]
```diff
@@ -1899,7 +1899,84 @@ public function sortedSetFetchByRank(string $cacheName, string $sortedSetName, ?
...
+     * Fetch the elements in a sorted set by score.
+     *
+     * @param string $cacheName Name of the cache that contains the set.
+     * @param string $sortedSetName The sorted set to fetch.
+     * @param ?float $minScore The minimum score (inclusive) of the
+     * elements to fetch. Defaults to negative infinity.
+     * @param ?float $maxScore The maximum score (inclusive) of the
+     * elements to fetch. Defaults to positive infinity.
+     * @param ?int $order The order to fetch the elements in. Defaults to Ascending.
+     *  Will be treated as ascending if SORT_ASC or higher is supplied, and descending if SORT_DESC or lower is supplied.
```
Comment (malandis) [inline src/Cache/CacheClient.php:1912]: What does "or higher" and "or lower" mean here?

File: src/Cache/Internal/ScsDataClient.php [hunk 2]
```diff
@@ -1701,6 +1702,89 @@ function () use ($call): SortedSetFetchResponse {
...
+
+            if (is_null($offset)) {
+                $byScore->setOffset(0);
+            } else {
+                $byScore->setOffset($offset);
+            }
+
+            if (is_null($count)) {
+                // a negative count returns all elements.
+                $byScore->setCount(-1);
```
Comment (malandis) [inline src/Cache/Internal/ScsDataClient.php:1746]: What about count 0?

Comment (nand4011) [reply src/Cache/Internal/ScsDataClient.php:1746]: Good catch.

**Review by malandis — APPROVED** [review]

File: src/Cache/CacheClient.php (hunk 1 above)
Comment (nand4011) [reply src/Cache/CacheClient.php:1912]: order is an integer. SORT_ASC is 4 and SORC_DESC is 3. you can supply any integer value. We're saying that if you supply 3 or less it is descending and if you supply 4 or more it is ascending.

**Pull request merged**